app/Services/ScrapeService.php

```diff
@@ -0,0 +1,227 @@
+        $buildingSelector = "[data-testid='accommodation-list-element']:nth-of-type($id)";
+        $timeout = config('scraper.crawler.timeouts.short');
+
+        $this->client->executeScript("document.querySelector(\"$buildingSelector\").scrollIntoView(true)");
+
+        $buildingElement = $this->client->waitFor($buildingSelector, $timeout)->filter($buildingSelector);
+
+        // Show photos panel
+        $buildingElement->filter("button")->click();
+
+        // Hotel name
+        $name = $this->fetchBuildingName($buildingElement);
+        // Hotel / hostel / resort etc.
+        $type = $this->fetchBuildingType($buildingElement);
+        $city = $this->fetchBuildingCity($buildingElement);
+
+        // Main image
+        $mainImage = $buildingElement->filter("[data-testid=\"accommodation-main-image\"]")->attr('src');
+
+        $this->client->waitFor("$buildingSelector [data-testid=\"grid-gallery\"]", $timeout);
+        $images = $this->fetchBuildingImages($buildingElement);
+
+        // Opens info panel
+        $buildingElement->filterXPath('.//button[contains(text(), "Info")]')->click();
+
+        // Description
+        $this->client->waitFor("$buildingSelector [data-testid=\"info-slideout\"]", $timeout);
+        $body = $this->fetchBuildingBody($buildingElement);
+        $street = $this->fetchBuildingAddress($buildingElement);
+
+        // More amenities button
+        $buildingElement->filter("[data-testid=\"toggle-all-amenities\"]")->click();
+
+        $amenities = $this->fetchBuildingAmenities($buildingElement);
+
+        return new Building(
+            $name,
+            $body,
+            $mainImage,
+            $city,
+            $street,
+            $amenities,
+            $images,
+            $type
+        );
+    }
+
+    private function appendToFiles(Building $building, string $directoryName): void
+    {
+        $imagesFilename = config('scraper.command.images_filename');
+        $imagesFile = fopen($directoryName .  $imagesFilename, 'a');
+
+        foreach ($building->images as $image) {
+            $record =  $building->name . ';' . $image;
+            fwrite($imagesFile, $record . "\n");
+        }
+        fclose($imagesFile);
+
+        $buildingsFilename = config('scraper.command.buildings_filename');
+        $buildingsFile = fopen($directoryName . $buildingsFilename, 'a');
+        $record = $building->name . ';' . $building->body . ';' . $building->street . ';'
+            . $building->city . ';' . $building->type . ';' . $building->mainImg;
+        fwrite($buildingsFile, $record . "\n");
+
+        fclose($buildingsFile);
+
+        $amenitiesFilename = config('scraper.command.amenities_filename');
+        $amenitiesFile = fopen($directoryName . $amenitiesFilename, 'a');
+
+        foreach ($building->amenities as $amenity) {
+            $record = $building->name . ';' . $amenity;
+            fwrite($amenitiesFile, $record . "\n");
+        }
+        fclose($amenitiesFile);
```

Comment:
Still looks like if you randomly added some new lines here and there.

My suggestion 1 (one with comments):
```php
        // Images
        $imagesFilename = config('scraper.command.images_filename');
        $imagesFile = fopen($directoryName .  $imagesFilename, 'a');

        foreach ($building->images as $image) {
            $record =  $building->name . ';' . $image;
            fwrite($imagesFile, $record . "\n");
        }

        fclose($imagesFile);

        // Buildings
        $buildingsFilename = config('scraper.command.buildings_filename');
        $buildingsFile = fopen($directoryName . $buildingsFilename, 'a');

        $record = $building->name . ';' . $building->body . ';' . $building->street . ';'
            . $building->city . ';' . $building->type . ';' . $building->mainImg;
        fwrite($buildingsFile, $record . "\n");

        fclose($buildingsFile);

        // Amenities
        $amenitiesFilename = config('scraper.command.amenities_filename');
        $amenitiesFile = fopen($directoryName . $amenitiesFilename, 'a');

        foreach ($building->amenities as $amenity) {
            $record = $building->name . ';' . $amenity;
            fwrite($amenitiesFile, $record . "\n");
        }

        fclose($amenitiesFile);
```

My suggestion 2 (the better one, IMO):
Just move every section to a different method and run `$this->appendImagesToFile(...)`, etc.